elastic/beats
# Move http_endpoint input to v2 input API

- Refactoring

## What does this PR do?

Update the http_endpoint input to be based on the v2 input API.

This change unexports most symbols and splits the HttpEndpoint into a httpEndpoint (config handling and server run), httpHandler, and a validator type.

The server used to set up a timeout of 10s, based on the period of the Run method gets called. With the v2 API the Run method must block until shutdown or on fatal error. We keep the server alive now.

The json object parsing is local to the handler method only now and not stored in the HTTPEndpoint anymore. This fixes a potential bug with events not getting published or duplicated if multiple clients are served concurrently.

The config validation for the authentication configuration was done in the validation step. This has been moved to `config.Validate` to stop the initialization process early if an invalid configuration has been provided.

Note: I mostly moved code around. The input still needs more attention (in follow up PRs) regarding testing, and code documentation. All in all the changes should not be user visible.

## Checklist



~~- [ ] I have commented my code, particularly in hard-to-understand areas~~
~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~
~~- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

## How to test this PR locally

configure the http_endpoint input and send JSON objects via curl. The objects should be visible in the beats output.

## Related issues


-
